content/docs/tutorials/certificate-defaults/README.md

```diff
@@ -0,0 +1,473 @@
+- Set defaults for all the `spec.privateKey` fields.
+
+1. Lets first take a look at the `ClusterPolicy`:
+
+    ```yaml file=./yamls/cpol-mutate-certificate-defaults.yaml
+    ```
+
+1. Apply the policy to the cluster and check it is ready:
+
+    ```shell
+    kubectl apply -f yamls/cpol-mutate-certificate-defaults.yaml
+    kubectl get cpol
+    ```
+    
+    When ready the output should look like this:
+
+    ```log
+    NAME                            BACKGROUND   VALIDATE ACTION   READY   AGE   MESSAGE
+    0-mutate-certificate-defaults   true         Audit             True    4s    Ready
+    ```
+
+1. Now inspect the `Certificate` for testing with the name "test-revision":
+
+    ```yaml file=./yamls/cert-test-revision.yaml
+    ```
+
+    You can see that we have set the most minimal configuration currently possible, specifying only a DNS name for the certificate, where to save it (`secretName`) and the issuer to use to request the certificate (`issuerRef`).
+
+1. Use the following command to dry-run apply the certificate and then `diff` it against the original resource, to see how the defaults from our `ClusterPolicy` are applied:
+
+    ```shell
+    kubectl apply -f yamls/cert-test-revision.yaml --dry-run=server -o yaml | diff -uZ yamls/cert-test-revision.yaml -
+    ```
+
+    The output should look similar to this:
+
+    ```shell
+    --- yamls/cert-test-revision.yaml	2023-09-07 16:58:48.975541922 +0100
+    +++ -	2023-09-08 09:16:02.151849668 +0100
+    @@ -1,8 +1,18 @@
+    apiVersion: cert-manager.io/v1
+    kind: Certificate
+    metadata:
+    +  annotations:
+    +    kubectl.kubernetes.io/last-applied-configuration: |
+    +      {"apiVersion":"cert-manager.io/v1","kind":"Certificate","metadata":{"annotations":{},"name":"test-revision","namespace":"default"},"spec":{"dnsNames":["example.com"],"issuerRef":{"group":"cert-manager.io","kind":"ClusterIssuer","name":"not-my-corp-issuer"},"secretName":"test-revision-cert"}}
+    +    policies.kyverno.io/last-applied-patches: |
+    +      set-privateKey-details.0-mutate-certificate-defaults.kyverno.io: added /spec/privateKey/encoding
+    +      set-privateKey-rotationPolicy.0-mutate-certificate-defaults.kyverno.io: added /spec/privateKey
+    +      set-revisionHistoryLimit.0-mutate-certificate-defaults.kyverno.io: added /spec/revisionHistoryLimit
+    +  creationTimestamp: "2023-09-08T08:16:02Z"
+    +  generation: 1
+      name: test-revision
+      namespace: default
+    +  uid: 88a9a28a-c354-4397-be62-7eee30104fc6
+    spec:
+      dnsNames:
+      - example.com
+    @@ -10,4 +20,10 @@
+        group: cert-manager.io
+        kind: ClusterIssuer
+        name: not-my-corp-issuer
+    +  privateKey:
+    +    algorithm: ECDSA
+    +    encoding: PKCS1
+    +    rotationPolicy: Always
+    +    size: 521
+    +  revisionHistoryLimit: 2
+      secretName: test-revision-cert
+    \ No newline at end of file
```

Comment:
```suggestion
```

Add a newline to the source file to avoid this warning.

Reply:
Caught this one, thank you.